lib/core/webdriver/firefox.js

```diff
@@ -211,6 +211,14 @@ module.exports.configureBuilder = function(builder, options) {
     profile.setPreference(pref, defaultFirefoxPreferences[pref]);
   });
 
+  if (options.firefox && options.firefox.preference) {
+    const userPrefs = Array.isArray(options.firefox.preference) ? options.firefox.preference : [options.firefox.preference];
```

Comment:
Could you extract ```toArray```from engine.js to a util model, and use that here too? Then you can skip the if statement altogether, just
```for (const pref of toArray(firefoxConfig.preference))```